internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -335,22 +335,20 @@ func (a *Adapter) annotateBuildPipelineRunWithSnapshot(snapshot *applicationapiv
 
 // updateBuildPipelineRunWithFinalInfo adds the final pieces of information to the pipelineRun in order to ensure
 // that anything that happened during the reconciliation is reflected in the CR
-func (a *Adapter) updateBuildPipelineRunWithFinalInfo(canRemoveFinalizer bool) error {
+func (a *Adapter) updateBuildPipelineRunWithFinalInfo(canRemoveFinalizer bool, cerr error) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		var err error
 		a.pipelineRun, err = a.loader.GetPipelineRun(a.context, a.client, a.pipelineRun.Name, a.pipelineRun.Namespace)
 		if err != nil {
 			return err
 		}
 
-		if a.pipelineRun.GetDeletionTimestamp() != nil {
-			err = tekton.AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(a.context, a.pipelineRun, a.client, err)
-			if err != nil {
-				return err
-			}
-			a.logger.LogAuditEvent("Updated build pipelineRun", a.pipelineRun, h.LogActionUpdate,
-				h.CreateSnapshotAnnotationName, a.pipelineRun.Annotations[h.CreateSnapshotAnnotationName])
+		err = tekton.AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(a.context, a.pipelineRun, a.client, cerr)
```

Comment:
Will annotating a `pipelineRun` that's being deleted work? Afaik, after the `deletionTimestamp` is set on a resource, only updates to the finalizers field are allowed.

Reply:
Thanks! Yes. I just think we should not have `a.pipelineRun.GetDeletionTimestamp() != nil`. So let me update it to `a.pipelineRun.GetDeletionTimestamp() == nil`

Reply:
Good point, looks good now, thanks!